third_party/amd/lib/TritonAMDGPUToLLVM/MaskedOpsToLLVM.cpp

```diff
@@ -0,0 +1,198 @@
+#include "AsyncUtility.h" // is this okay?
+#include "Dialect/TritonAMDGPU/IR/Dialect.h"
+#include "PatternTritonGPUOpToLLVM.h"
+#include "TritonAMDGPUToLLVM/Passes.h"
+#include "Utility.h"
+#include "mlir/Conversion/LLVMCommon/TypeConverter.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "triton/Tools/Sys/GetEnv.hpp"
+#include <optional>
+
+namespace mlir::triton {
+#define GEN_PASS_DEF_CONVERTMASKEDOPSTOLLVM
+#include "TritonAMDGPUToLLVM/Passes.h.inc"
+} // namespace mlir::triton
+
+using namespace mlir;
+using namespace mlir::triton::gpu;
+
+namespace {
+
+static std::pair<bool, bool>
+getCacheModifierFlagsForLoad(triton::amdgpu::MaskedLoadOp loadOp) {
+  auto cm = loadOp.getCache();
+  bool isVolatile = false;
+  bool isNonTemporal = false;
+
+  switch (cm) {
+  case triton::CacheModifier::CA:
+    // ca: volatile=false, nontemporal=false
+    break;
+  case triton::CacheModifier::CG:
+    // cg: volatile=false, nontemporal=true
+    isNonTemporal = true;
+    break;
+  case triton::CacheModifier::CV:
+    // cv: volatile=true, nontemporal=X
+    isVolatile = true;
```

Comment:
This also makes nontemporal as true in the old place? Can we be consistent without silently changing behavior?